### PR TITLE
Make ExternalCompactionIT faster by reusing MiniAccumuloCluster

### LINKED ISSUE #2159: Review for unused ExternalCompaction methods
As part of writing documentation for a future blog post I found that Compactor.cancel() and CompactionCoordinator.getCompactorConnection() might not be referenced anywhere.

### PR DESCRIPTION
**Pull request by dlmarion**
Modified the test to use a different parent class that uses a
shared MiniAccumuloCluster for the different tests. Also reduced
the time based configuration properties in an attempt to make
things faster.

Closes #2159

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java [hunk 1]
```diff
@@ -193,22 +195,55 @@ public boolean accept(Key k, Value v) {
...
+        .from(SharedMiniClusterBase.getCluster().getClientProperties()).build()) {
+      for (String table : tables) {
+        try {
+          client.tableOperations().delete(table);
+        } catch (Exception e) {
+          fail("Error deleting table: " + table + ", msg: " + e.getMessage());
+        }
+      }
+    }
+  }
```
Comment (milleruntime) [inline test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:226]: Looks like you have a unique table name per test. If that is the case do you need to delete them after each test?

**Review by milleruntime — APPROVED** [review]
Assuming it is faster, looks good. I thought you may be able to save time if you don't care about cleaning up tables between tests. Another thing to consider is the new reuseForks behavior in our build. This seems like a test where we may not want to reuse forks.

File: test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java [hunk 2]
```diff
@@ -193,22 +195,55 @@ public boolean accept(Key k, Value v) {
...
   }
 
+  @Before
+  public void setUp() throws Exception {
+    if (SharedMiniClusterBase.getCluster() == null) {
+      SharedMiniClusterBase.startMiniClusterWithConfig(this);
+    }
+  }
+
+  private void stopProcess(Process process) throws Exception {
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:205]: I saw this is called several times in some places with the same pattern. Could do the null check inside this method, and the arg list could be varargs: ```private void stopProcess(ProcessInfo ... processes) ... ```

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:205]: Yes, good idea.

File: test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java (hunk 1 above)
Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:226]: I do need to delete them, or maybe change some other code. There are places in the tests where I am checking the external compaction final state markers in the metadata table and I'm expecting one entry. If I don't delete the table, then I end up with more than one. IIRC there is not a tableId in the final state marker for me to use to verify that it's complete. I'll double-check.

Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:226]: Should/can the tableId be added to the final state marker?

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:226]: So, I just spent hours trying to figure out why a test was failing to complete after switching the IT to use a different parent class that does not restart MAC between tests. The test was running a compaction from the previous test due to the way the previous test ended. When the compactor and coordinator started for the current test the tablet server gave it a compaction from the prior test. I guess I should be happy that it's resilient to components being restarted, but I was banging my head against the wall. For the purposes of the IT, I have to delete the tables between tests.

File: test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java (hunk 2 above)
Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java]: Addressed in da6017f

File: test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java (hunk 1 above)
Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ExternalCompactionIT.java:244]: Addressed in da6017f by deleting tables in tearDown() after each test.

**Review by ctubbsii — APPROVED** [review]

**Pull request merged**